system/datatypes/tagindex.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/*
+	Class: tagindexpage
+	Shows an index page for articles based on their tags
+*/
+	class tagindexpage extends HyphaSystemPage {
+		function __construct(RequestContext $O_O) {
+			parent::__construct($O_O);
+		}
+
+		public function process(HyphaRequest $request) {
+			return $this->tagIndexView($request);
+		}
+
+		public function tagIndexView(HyphaRequest $request) {
+			$lang = $request->getArg(0);
+			// TODO: This urldecode should really be done by
+			// HyphaRequest (#274)
+			$label = urldecode($request->getArg(1));
+
+			$tag = HyphaTags::findTagByLabel($lang, $label);
+			if ($tag === null)
+				return '404';
+
+			/** @var HyphaDomElement $main */
+			$main = $this->html->find('#main');
+
+			/** @var HyphaDomElement $pagename */
+			$pagename = $this->html->find('#pagename');
+			$tagprefixspan = $this->html->createElement('span');
+			$tagprefixspan->setText('Index for tag: ');
+			$tagprefixspan->setAttribute('class', 'tag_prefix');
+			$tagspan = $this->html->createElement('span');
+			$tagspan->setAttribute('class', 'selected_tag');
+			$tagspan->setText($label);
+			$pagename->append($tagprefixspan);
+			$pagename->append($tagspan);
+
+			$includePrivate = isUser();
+			$pages = $this->getPages($tag, $includePrivate);
+			$pages = $this->sortPages($pages);
+
+			$pages_list = $this->renderPages($pages);
+			$main->append($pages_list);
+		}
+
+		private function sortPages($pages) {
+			// TODO: This should cache sort dates
+			$compare = function($a, $b) {
+				$a_date = $a->getSortDateTime();
+				$b_date = $b->getSortDateTime();
+				// TODO Can we better handle null here?
+				return ($b_date ? $b_date->getTimestamp() : 0) - ($a_date ? $a_date->getTimestamp() : 0);
+			};
+			usort($pages, $compare);
+			return $pages;
+		}
+
+		private function getPages(HyphaTag $tag, $includePrivate = false) {
+			$pagetypes = ['peer_reviewed_article'];
+			$pagetypes = [];
```

Comment:
variable is overwritten here

Reply:
Ah, that was old code, I removed the first line.